Draft of issues fixed (#129)
idontgetoutmuch/random: CHANGELOG.md

9. [Continuous integration](https://travis-ci.org/github/idontgetoutmuch/random).
10. Fully documented - for more details see the [haddock](https://htmlpreview.github.io/?https://github.com/idontgetoutmuch/random/blob/release-notes/docs/System-Random.html).

### Issues Addressed

 Issue Number | Description | Comment
--------------|-------------|--------

>  [25](https://github.com/haskell/random/issues/25) | The seeds generated by split are not independent |

Reviewer: ```suggestion
 [25](https://github.com/haskell/random/issues/25) | The seeds generated by split are not independent | Changed algorithm to SplitMix, which provides a robust split operation
```

>  [26](https://github.com/haskell/random/issues/26) | Add Random instances for tuples |

Reviewer: ```suggestion
 [26](https://github.com/haskell/random/issues/26) | Add Random instances for tuples | Added Uniform instances for tuples
```

>  [44](https://github.com/haskell/random/issues/44) | Add Random instance for Natural |

Reviewer: ```suggestion
 [44](https://github.com/haskell/random/issues/44) | Add Random instance for Natural | Added UniformRange instance for Natural
```

>  [51](https://github.com/haskell/random/issues/51) | Very low throughput |

Reviewer: ```suggestion
 [51](https://github.com/haskell/random/issues/51) | Very low throughput | Much higher throughput - see benchmarks below
```

>  [53](https://github.com/haskell/random/issues/53) | incorrect distribution of randomR for floating-point numbers | Fixed

Reviewer: ```suggestion
 [53](https://github.com/haskell/random/issues/53) | incorrect distribution of randomR for floating-point numbers | Documented behaviour of floating point numbers
```

We haven't actually _done_ this yet :) I'll create a PR.

Author: Ooh yes we meant to do it - thanks

Reviewer: I've started writing a little Haddock section for this. I'll create a PR once I have a draft.

>  [55](https://github.com/haskell/random/issues/55) | System/Random.hs:43:1: warning: [-Wtabs] |

Reviewer: ```suggestion
 [55](https://github.com/haskell/random/issues/55) | System/Random.hs:43:1: warning: [-Wtabs] | No more tabs
```

Perhaps this comment isn't necessary ...

>  [58](https://github.com/haskell/random/issues/58) | Why does random for Float and Double produce exactly 24 or 53 bits? | (*)

Reviewer: I'm not sure we have really fixed this. The division method provides more coverage than putting 23 bits into the mantissa as was done before. But we haven't quantified the gain yet. The comment about sampling all floating point numbers probably doesn't help people who don't have all the context very much. So in a sense in https://github.com/idontgetoutmuch/random/issues/113#issuecomment-624041080 we have decided not to tackle this issue in v1.2.

Author: I was careful to say issues "addressed" and then put fixed in the comment section for those that really are fixed. I think we should note that 1.2 provides more bits and to look at https://github.com/idontgetoutmuch/random/issues/105 for more discussion.

What do you think? Should this only be issues closed?

Reviewer: Perhaps it is worth making the different ways in which we've addressed issues more explicit? I'm just thinking out loud here.

- Issues 25 and 51 are now actually fixed. They motivated this entire endeavour, and we've succeeded. So separating them out may not be a bad idea. (I would also say that issue 55 has been fixed, but it's less important.)
- The remaining issues apart from 53 and 58 I would say have been "addressed": e.g. issue 44 asks for a `Random` instance for `Natural`; we've implemented a `UniformRange` instance instead. So these are issues we've taken on board and done the closest thing to fixing them in the new framework.
- 53 and 58 are in the third group of "things we've though hard about". We have a better answer for why things are the way they are, but haven't changed the actual behaviour all that much. Perhaps these could be filed under "documentation improvements" or "API clarifications" or something like that.

Author: How about tagging them  in the comments column as fixed / addressed / clarifications?

>  [59](https://github.com/haskell/random/issues/59) | read :: StdGen fails for strings longer than 6 |

Reviewer: ```suggestion
 [59](https://github.com/haskell/random/issues/59) | read :: StdGen fails for strings longer than 6 | StdGen is no longer an instance of Read
```

9. [Continuous integration](https://travis-ci.org/github/idontgetoutmuch/random).
10. Fully documented - for more details see the [haddock](https://htmlpreview.github.io/?https://github.com/idontgetoutmuch/random/blob/release-notes/docs/System-Random.html).

### Issues Addressed

 Issue Number | Description | Comment
--------------|-------------|--------
 [25](https://github.com/haskell/random/issues/25) | The seeds generated by split are not independent | Fixed: changed algorithm to SplitMix, which provides a robust split operation
 [26](https://github.com/haskell/random/issues/26) | Add Random instances for tuples | Addressed: added `Uniform` instances for up to 6-tuples
 [44](https://github.com/haskell/random/issues/44) | Add Random instance for Natural | Addressed: added UniformRange instance for Natural
 [51](https://github.com/haskell/random/issues/51) | Very low throughput | Fixed: see benchmarks below

>  [53](https://github.com/haskell/random/issues/53) | incorrect distribution of randomR for floating-point numbers | (*)

Reviewer: ```suggestion
 [53](https://github.com/haskell/random/issues/53) | incorrect distribution of randomR for floating-point numbers | (\*)
```

`*` starts italic formatting. A backslash escapes it, so editors don't get confused: https://daringfireball.net/projects/markdown/syntax#backslash

 [55](https://github.com/haskell/random/issues/55) | System/Random.hs:43:1: warning: [-Wtabs] | Fixed: No more tabs

>  [58](https://github.com/haskell/random/issues/58) | Why does random for Float and Double produce exactly 24 or 53 bits? | (*)

Reviewer: ```suggestion
 [58](https://github.com/haskell/random/issues/58) | Why does random for Float and Double produce exactly 24 or 53 bits? | (\*)
```
(see above)

 [59](https://github.com/haskell/random/issues/59) | read :: StdGen fails for strings longer than 6 | Addressed: StdGen is no longer an instance of Read

#### Comments

> (*) 1.2 samples more bits but does not sample every `Float` or

Reviewer: ```suggestion
(\*) 1.2 samples more bits but does not sample every `Float` or
```
(see above)
